src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -342,5 +361,54 @@ private static void SetConsoleOutputEncoding(System.Text.Encoding encoding)
             {
             }
         }
+
+        private static void LogHelperMessageForPathTooLongException(Console logger)
+        {
+            if (!IsWindows10(logger))
+            {
+                logger.WriteWarning(LocalizedResourceManager.GetString(nameof(NuGetResources.Warning_LongPath_UnsupportOS)));
+            }
+            else if (!IsSupportLongPathEnabled(logger))
+            {
+                logger.WriteWarning(LocalizedResourceManager.GetString(nameof(NuGetResources.Warning_LongPath_DisabledPolicy)));
+            }
+            else if (!IsRuntimeGreaterThanNet462(logger))
+            {
+                logger.WriteWarning(LocalizedResourceManager.GetString(nameof(NuGetResources.Warning_LongPath_UnsupportedNetFramework)));
+            }
+        }
+
+        private static bool IsWindows10(ILogger logger)
+        {
+            var reg = RegistryKeyUtility.OpenSubKey(OSVersionRegistryKey, Registry.LocalMachine, logger);
```

Comment:
I'd expect it to open a regKey, read a specific value and close it right away. The current implementation leaves a loophole of keeping these regKeys open forever if there were intermittent exceptions.